src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp

```diff
@@ -1819,6 +1878,35 @@ class spell_halion_spawn_living_embers : public SpellScriptLoader
         }
 };
 
+class spell_halion_blazing_aura : public SpellScriptLoader
+{
+    public:
+        spell_halion_blazing_aura() : SpellScriptLoader("spell_halion_blazing_aura") { }
+
+        class spell_halion_blazing_aura_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_halion_blazing_aura_SpellScript);
+
+            void HandleScript(SpellEffIndex effIndex)
+            {
+                PreventHitDefaultEffect(effIndex);
+                GetHitUnit()->CastSpell(GetHitUnit(), GetSpellInfo()->Effects[EFFECT_1].TriggerSpell);
```

Comment:
Well, this is the only way I could find to make ir work and there's similar things over the core, open to suggestions tho